### ISSUE TITLE
feature/chat branch: Typed text dissappears !

### ISSUE THREAD

**Issue by aelmahmoudy** [redsolution/xabber-android#338]
Hello,

  I understand that this isn't complete yet. I was trying this commit: https://github.com/redsolution/xabber-android/commit/f24bde0c8cc601cd2cef1de7a8c572cd8d6f7add

I found that when I type text, it sometimes dissappears before I press send (or even during the typing !). By checking the logcat, I found that the text dissappears because another XMPP account failed to connect:
```
W/System.err(  705): com.xabber.android.data.account.AccountItem@41d98a08:aelmahmoudy@mycompany.com/androidmrUC1Te0
W/System.err(  705): java.lang.RuntimeException: XMPPError connecting to ip.ad.re.es:5222.: remote-server-error(502) XMPPError connecting to ip.ad.re.es:5222.
W/System.err(  705):   -- caused by: java.net.ConnectException: failed to connect to /ip.ad.re.es (port 5222): connect failed: ETIMEDOUT (Connection timed out)
W/System.err(  705):    at com.xabber.android.data.connection.ConnectionThread.connect(ConnectionThread.java:424)
W/System.err(  705):    at com.xabber.android.data.connection.ConnectionThread.access$300(ConnectionThread.java:55)
W/System.err(  705):    at com.xabber.android.data.connection.ConnectionThread$7.run(ConnectionThread.java:331)
W/System.err(  705):    at com.xabber.android.data.connection.ConnectionThread$20.run(ConnectionThread.java:660)
W/System.err(  705):    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
W/System.err(  705):    at java.util.concurrent.FutureTask.run(FutureTask.java:237)
W/System.err(  705):    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
W/System.err(  705):    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
W/System.err(  705):    at java.lang.Thread.run(Thread.java:841)
W/System.err(  705): Caused by: XMPPError connecting to ip.ad.re.es:5222.: remote-server-error(502) XMPPError connecting to 173.254.28.46:5222.
W/System.err(  705):   -- caused by: java.net.ConnectException: failed to connect to /ip.ad.re.es (port 5222): connect failed: ETIMEDOUT (Connection timed out)
W/System.err(  705):    at org.jivesoftware.smack.XMPPConnection.connectUsingConfiguration(XMPPConnection.java:575)
W/System.err(  705):    at org.jivesoftware.smack.XMPPConnection.connect(XMPPConnection.java:1018)
W/System.err(  705):    at com.xabber.android.data.connection.ConnectionThread.connect(ConnectionThread.java:419)
W/System.err(  705):    ... 8 more
I/com.xabber.android.data.account.AccountItem@41d98a08:aelmahmoudy@mycompany.com/androidmrUC1Te0(  705): onClose 1107311960 - 140, true
I/com.xabber.android.ui.adapter.ChatViewerAdapter@4256e4b8(  705): onChange: 
I/com.xabber.android.ui.adapter.ChatViewerAdapter@4256e4b8(  705): setPrimaryItem position: 0
I/com.xabber.android.data.notification.NotificationManager@41d70278(  705): Notification: 1, ticker: null, sound: null, vibro: 0, light: 0
I/com.xabber.android.ui.adapter.ChatViewerAdapter@4256e4b8(  705): setPrimaryItem position: 0
```

**Comment by aelmahmoudy** [comment]
A missing feature with that new ChatViewer is that single messages could no more be quoted/copied.

**Comment by grigoryfedorov** [comment]
Thanks a lot!

**Comment by grigoryfedorov** [comment]
@aelmahmoudy ChatViewer context menu fixed in 4ba3b54a6223240cdf352b245463f788dbd9f361

**Comment by grigoryfedorov** [comment]
@aelmahmoudy input text dissappearing fixed in 3bdcd2050f5b3817c0da6dec93147694e62b30a5

**Comment by aelmahmoudy** [comment]
Thanks
